src/libraries/System.Console/src/System/ConsolePal.TerminalFormatStrings.cs

```diff
@@ -0,0 +1,255 @@
+        public readonly string? ClrEol;
+        /// <summary>The ANSI-compatible string for the Cursor Position report request.</summary>
+        /// <remarks>
+        /// This should really be in user string 7 in the terminfo file, but some terminfo databases
+        /// are missing it.  As this is defined to be supported by any ANSI-compatible terminal,
+        /// we assume it's available; doing so means CursorTop/Left will work even if the terminfo database
+        /// doesn't contain it (as appears to be the case with e.g. screen and tmux on Ubuntu), at the risk
+        /// of outputting the sequence on some terminal that's not compatible.
+        /// </remarks>
+        public const string CursorPositionReport = "\x1B[6n";
+        /// <summary>
+        /// The dictionary of keystring to ConsoleKeyInfo.
+        /// Only some members of the ConsoleKeyInfo are used; in particular, the actual char is ignored.
+        /// </summary>
+        public readonly Dictionary<ReadOnlyMemory<char>, ConsoleKeyInfo> KeyFormatToConsoleKey =
+            new Dictionary<ReadOnlyMemory<char>, ConsoleKeyInfo>(new ReadOnlyMemoryContentComparer());
+
+        /// <summary> Max key length </summary>
+        public readonly int MaxKeyFormatLength;
+        /// <summary> Min key length </summary>
+        public readonly int MinKeyFormatLength;
+        /// <summary>The ANSI string used to enter "application" / "keypad transmit" mode.</summary>
+        public readonly string? KeypadXmit;
+        /// <summary>Indicates that it was created out of rxvt TERM</summary>
+        public readonly bool IsRxvtTerm;
+
+        public TerminalFormatStrings(TermInfo.Database? db)
+        {
+            if (db == null)
+                return;
+
+            KeypadXmit = db.GetString(TermInfo.WellKnownStrings.KeypadXmit);
+            Foreground = db.GetString(TermInfo.WellKnownStrings.SetAnsiForeground);
+            Background = db.GetString(TermInfo.WellKnownStrings.SetAnsiBackground);
+            Reset = db.GetString(TermInfo.WellKnownStrings.OrigPairs) ?? db.GetString(TermInfo.WellKnownStrings.OrigColors);
+            Bell = db.GetString(TermInfo.WellKnownStrings.Bell);
+            Clear = db.GetString(TermInfo.WellKnownStrings.Clear);
+            Columns = db.GetNumber(TermInfo.WellKnownNumbers.Columns);
+            Lines = db.GetNumber(TermInfo.WellKnownNumbers.Lines);
+            CursorVisible = db.GetString(TermInfo.WellKnownStrings.CursorVisible);
+            CursorInvisible = db.GetString(TermInfo.WellKnownStrings.CursorInvisible);
+            CursorAddress = db.GetString(TermInfo.WellKnownStrings.CursorAddress);
+            CursorLeft = db.GetString(TermInfo.WellKnownStrings.CursorLeft);
+            ClrEol = db.GetString(TermInfo.WellKnownStrings.ClrEol);
+
+            IsRxvtTerm = !string.IsNullOrEmpty(db.Term) && db.Term.Contains("rxvt", StringComparison.OrdinalIgnoreCase);
```

Comment:
this is the only change I've made to this file (introducing an initializing `IsRxvtTerm` property). The rest was just moved from `TermInfo.cs`